packages/db/src/models/granule.ts

```diff
@@ -128,6 +129,42 @@ export default class GranulePgModel extends BasePgModel<PostgresGranule, Postgre
       .where(knexOrTrx.raw(`${this.tableName}.created_at <= to_timestamp(${translateDateToUTC(granule.created_at)})`))
       .returning('cumulus_id');
   }
+
+  /**
+   * Get granules from the granule cumulus_id
+   *
+   * @param {Knex | Knex.Transaction} knexOrTrx -
+   *  DB client or transaction
+   * @param {Array<number>} granuleCumulusIds -
+   * single granule cumulus_id or array of granule cumulus_ids
+   * @param {Object} [params] - Optional object with addition params for query
+   * @param {number} [params.limit] - number of records to be returned
+   * @param {number} [params.offset] - record offset
+   * @returns {Promise<Array<number>>} An array of granules
+   */
+  async searchByCumulusIds(
+    knexOrTrx: Knex | Knex.Transaction,
+    granuleCumulusIds: Array<number> | number,
+    params: { limit: number, offset: number }
+  ): Promise<Array<number>> {
+    const { limit, offset, ...sortQueries } = params || {};
+    const sortFields = getSortFields(sortQueries);
+    const granuleCumulusIdsArray = [granuleCumulusIds].flat();
+    const granules = await knexOrTrx(this.tableName)
+      .whereIn('cumulus_id', granuleCumulusIdsArray)
+      .modify((queryBuilder) => {
+        if (limit) queryBuilder.limit(limit);
+        if (offset) queryBuilder.offset(offset);
+        if (sortFields.length >= 1) {
+          sortFields.forEach((sortObject: { [key: string]: { order: string } }) => {
+            const sortField = Object.keys(sortObject)[0];
+            const { order } = sortObject[sortField];
+            queryBuilder.orderBy(sortField, order);
+          });
+        }
+      });
+    return granules;
+  }
```

Comment:
This method was pulled from Rds-Phase2 in support of the move of translatePostgresGranuleToApiGranule()